### DISCUSSION TITLE
Projecting different data set onto an existing UMAP?

### DISCUSSION THREAD

**Discussion by jogiles** [GreenleafLab/ArchR#573] [Feature Requests]
Hello!

Is it possible to use the predictive capability of UMAP? I would like to project a different data set onto an existing UMAP from an ArchR project. 

Thank you! Josephine

**Comment by rcorces** [comment]
This is definitely in the works and may already be available but it is not documented. @jgranja24 can provide more info.

**Comment by jgranja24** [comment]
Hi @jogiles,

This is something we are hoping to have out in the near future. Sorry I cant offer a more precise timeline!

**Comment by armonazizi** [comment]
Hey @rcorces and @jgranja24, thanks again for all the work you're putting into this. Has anything like what was mentioned above been implemented? I was searching around and saw that the `projectBulkATAC` function might do something similar but for bulk datasets; could that function be utilized for sets of single cells?

Best,
Armon

**Reply by rcorces** [reply]
There is no function implemented to do this but the `addUMAP()` function has the `saveModel` parameter which allows you to save the UMAP model to use for projection of other cells. I dont have code to help you do this but it is possible.

**Reply by armonazizi** [reply]
Thanks! will give this a try
